File: frontend/src/services/team.service.ts
import api from './api';
import {
  Team,
  TeamMember,
  CreateTeamRequest,
  UpdateTeamRequest,
} from '@/types/team.types';
import { Page } from '@/types/project.types';

export const teamService = {
  async create(data: CreateTeamRequest): Promise<Team> {
    const response = await api.post<Team>('/teams', data);
    return response.data;
  },

  async getById(id: string): Promise<Team> {
    const response = await api.get<Team>(`/teams/${id}`);
    return response.data;
  },

  async getMyTeams(page = 0, size = 20): Promise<Page<Team>> {
    const response = await api.get<Page<Team>>('/teams/my', {
      params: { page, size },
    });
    return response.data;
  },

  async getPublicTeams(page = 0, size = 20): Promise<Page<Team>> {
    const response = await api.get<Page<Team>>('/teams/public', {
      params: { page, size },
    });
    return response.data;
  },

  async update(id: string, data: UpdateTeamRequest): Promise<Team> {
    const response = await api.put<Team>(`/teams/${id}`, data);
    return response.data;
  },

  async delete(id: string): Promise<void> {
    await api.delete(`/teams/${id}`);
  },

  // Member management
  async getMembers(teamId: string): Promise<TeamMember[]> {
    const response = await api.get<TeamMember[]>(`/teams/${teamId}/members`);
    return response.data;
  },

  async addMember(teamId: string, userId: string, role = 'MEMBER'): Promise<TeamMember> {
    const response = await api.post<TeamMember>(`/teams/${teamId}/members/${userId}`, null, {
      params: { role },
    });
    return response.data;
  },

  async removeMember(teamId: string, userId: string): Promise<void> {
    await api.delete(`/teams/${teamId}/members/${userId}`);
  },

  async updateMemberRole(teamId: string, userId: string, role: string): Promise<void> {
    await api.put(`/teams/${teamId}/members/${userId}/role`, null, {
      params: { role },
    });
  },
};

File: backend/src/main/java/kg/taskflow/service/TeamService.java
package kg.taskflow.service;

import kg.taskflow.dto.team.*;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.UUID;

public interface TeamService {

    TeamDto create(CreateTeamRequest request);

    TeamDto getById(UUID id);

    Page<TeamDto> getMyTeams(Pageable pageable);

    Page<TeamDto> getPublicTeams(Pageable pageable);

    TeamDto update(UUID id, UpdateTeamRequest request);

    void delete(UUID id);

    // Member management
    TeamMemberDto addMember(UUID teamId, UUID userId, String role);

    void removeMember(UUID teamId, UUID userId);

    void updateMemberRole(UUID teamId, UUID userId, String role);

    List<TeamMemberDto> getMembers(UUID teamId);

    // Access checks
    boolean isMember(UUID teamId, UUID userId);

    boolean isAdmin(UUID teamId, UUID userId);
}
